linera-client/src/chain_listener.rs

```diff
@@ -196,6 +196,13 @@ impl<C: ClientContext> ChainListener<C> {
                 self.update_validators(&notification).await?;
                 self.update_wallet(notification.chain_id).await?;
                 self.add_new_chains(*hash).await?;
+                let publishers = self
+                    .update_event_subscriptions(notification.chain_id)
+                    .await?;
+                if !publishers.is_empty() {
```

Comment:
This if shouldn't be necessary - if `publishers` is empty then `listen_recursively` will return early.

Reply:
That's true. But maybe we should still avoid needlessly calling `process_inbox`.